Library/Homebrew/cmd/update.sh

```diff
@@ -378,19 +381,34 @@ EOS
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
+    cd "$DIR" || continue
+
+    TAP_VAR="$(repo_var "$DIR")"
+    declare PREFETCH_REVISION"$TAP_VAR"="$(read_current_revision)"
```

Comment:
Apologies for the obviously too hasty :+1: as this was a bug I should have been able to spot, but I got too caught up in some minor details and the `test-bot` discussion. 😭

Given how important a functioning `brew update` is for the rest of Homebrew, I think the morale of the story and how we approach future changes to the update should be:

- Don't rush `brew update` changes that aren't urgent bug fixes. (This was about 6 hours from an almost complete rewrite to merging this. Probably too quick.)
- Self-review your own changes after not looking at them for 24 hours.
- If possible, give more co-maintainers a chance to review.
- Do more local testing. If non-obvious, provide instructions how to test this locally, including what you think should happen in various scenarios.

I'm not blaming anyone here. Mistakes happen, but given this incident let's try to learn from them. `brew update` is way too critical to treat changes to it like any other PR.